pkg/controller/backupstoragelocation_controller.go

```diff
@@ -0,0 +1,147 @@
+type BackupStorageLocationReconciler struct {
+	Scheme          *runtime.Scheme
+	StorageLocation velero.StorageLocation
+
+	Log logrus.FieldLogger
+}
+
+// +kubebuilder:rbac:groups=velero.io,resources=backupstoragelocations,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=velero.io,resources=backupstoragelocations/status,verbs=get;update;patch
+func (r *BackupStorageLocationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithField("controller", "backupstoragelocation")
+
+	log.Info("Checking for existing backup locations ready to be verified; there needs to be at least 1 backup location available")
+
+	locationList, err := velero.ListBackupStorageLocations(r.StorageLocation.Client, r.StorageLocation.Ctx, req.Namespace)
+	if err != nil {
+		log.WithError(err).Error("No backup storage locations found, at least one is required")
+	}
+
+	if r.StorageLocation.DefaultStoreValidationFrequency <= 0 {
+		r.StorageLocation.DefaultStoreValidationFrequency = time.Minute
+	}
+
+	var defaultFound bool
+	var unavailableErrors []string
+	var anyVerified bool
+	for i := range locationList.Items {
+		location := &locationList.Items[i]
+		log := r.Log.WithField("controller", "backupstoragelocation").WithField("backupstoragelocation", location.Name)
+
+		if location.Name == r.StorageLocation.DefaultStorageLocation {
+			defaultFound = true
+		}
+
+		if !r.StorageLocation.IsReadyToValidate(location, log) {
+			continue
+		}
+
+		anyVerified = true
+
+		log.Debug("Verifying backup storage location")
+
+		if err := r.StorageLocation.IsValid(location, log); err != nil {
+			log.Debug("Backup location verified, not valid")
+			unavailableErrors = append(unavailableErrors, errors.Wrapf(err, "Backup location %q is unavailable", location.Name).Error())
+
+			if location.Name == r.StorageLocation.DefaultStorageLocation {
+				log.Warnf("The specified default backup location named %q is unavailable; for convenience, be sure to configure it properly or make another backup location that is available the default", r.StorageLocation.DefaultStorageLocation)
+			}
+
+			if err2 := r.StorageLocation.PatchStatus(location, velerov1api.BackupStorageLocationPhaseUnavailable); err2 != nil {
+				log.WithError(err).Errorf("Error updating backup location phase to %s", velerov1api.BackupStorageLocationPhaseUnavailable)
+				continue
+			}
+		} else {
+			log.Debug("Backup location verified and it is valid")
+			if err := r.StorageLocation.PatchStatus(location, velerov1api.BackupStorageLocationPhaseAvailable); err != nil {
+				log.WithError(err).Errorf("Error updating backup location phase to %s", velerov1api.BackupStorageLocationPhaseAvailable)
+				continue
+			}
+		}
+	}
+
+	if !anyVerified {
+		log.Info("No backup locations were ready to be verified")
+	}
+
+	r.logReconciledPhase(defaultFound, locationList, unavailableErrors)
+
+	return ctrl.Result{Requeue: true}, nil
```

Comment:
`Requeue` makes the controller requeue the request, which we want to do only if we run into errors reconciling the request and we want to retry.
In a success scenario, we want to return `ctrl.Result{}, nil`
https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.5.2/pkg/reconcile?tab=doc#Result

Reply:
Actually no, in this case we do want to requeue continuously. We have an external dependency (storage) that can change state at any time and this is what we are continuously reconciling.